src/main/java/uk/gov/justice/hmpps/prison/repository/InmateRepository.java

```diff
@@ -610,6 +609,14 @@ private Long getCategoryAssessmentTypeId() {
                 .collect(Collectors.toList());
     }
 
+    private List<OffenderCategorise> removeOffendersWithNoPreviousStandardCategorisations(final List<OffenderCategorise> catList) {
+        final var standardCategoryCodes = Set.of("B", "C", "D");
+        final var offenderNoMap = catList.stream().collect(Collectors.groupingBy(OffenderCategorise::getOffenderNo));
+        return offenderNoMap.values().stream().filter(oc ->
```

Comment:
thinking about it again this morning, should this be offenderNo or bookingId?

Reply:
We should remove the offender if no std cat in the current bookingId. But as it happens getOffenderNo is synonymous here as we should only get the current booking from the db for each offender (as they can only be 'in prison' for one booking)

Reply:
Another thought is that we should probably do all this in one strean pipeline !

Reply:
Brill cheers thought I'd double check, yeah thought a bit a refactoring might be needed! 😁

Reply:
👍